### DISCUSSION TITLE
Change default type mapping in generated schema

### DISCUSSION THREAD

**Discussion by securityMB** [mikro-orm/mikro-orm#1586] [Q&A]
Hello!

By default, when using `SchemaGenerator`, properties typed as `string` are created as columns of type `varchar(255)`, as in the example:

```typescript
@Entity()
export class Test {
  @Property()
  test: string; // this will be varchar(255);
}
```

Is it possible to change the default to `text` considering that in Postgres using `varchar(255)` makes [no sense](https://stackoverflow.com/questions/4848964/difference-between-text-and-varchar-character-varying/4849030#4849030)?

I'm aware I can change the type with `{columnType: 'text'}` but I'm thinking about doing it globally.

**Comment by securityMB** [comment]
Okay, so I added a very silly hack just before loading mikro-orm:

```typescript
import {PostgreSqlDriver, PostgreSqlSchemaHelper} from "@mikro-orm/postgresql";
// (...)
PostgreSqlSchemaHelper.TYPES.string = ["text"];
```

But I wish there was a better way.

**Comment by B4nan** [comment]
You could create custom driver by extending the current one and overriding the `SchemaHelper` where this is handled.
